### PR TITLE
feat(ext/ffi): support passing and returning bigints

### LINKED ISSUE #11641: ffi panic on integer parameters over 32bits
I've been playing around with the new `Deno.dlopen`. In doing so, I've noticed that integers larger than 32bits produce some weird quirks.
Here is some test C code...
```c
#include <stdint.h>
void takeU64(uint64_t x) {}
void takeU32(uint32_t x) {}
```
and a `dlopen`...
```ts
const dylib = Deno.dlopen("usize.dll", {
  "takeU64": { parameters: ["u64"], result: "void" },
  "takeU32": { parameters: ["u32"], result: "void" },
});
```

And some calls to demonstrate the issue:
```ts
// Perhaps expected, but C would truncate: (uint32_t)(8589934591) == 4294967295
dylib.symbols.takeU32(0x00000001FFFFFFFF);
// thread 'main' panicked at 'Expected ffi arg value to be an unsigned integer', extensions\ffi\lib.rs:260:6

// This runs successfully.
dylib.symbols.takeU64(0x00000000FFFFFFFF);
// This does not.
dylib.symbols.takeU64(0x00000001FFFFFFFF);
// thread 'main' panicked at 'Expected ffi arg value to be an unsigned integer', extensions\ffi\lib.rs:260:6
```

---

I also tried using BigInts to see if this was a problem I could avoid todo with JavaScript's 2^53 integer limit, but this also caused a panic:
```ts
dylib.symbols.takeU64(BigInt(0x00000001FFFFFFFF));
// thread 'main' panicked at 'serde_v8: unknown ValueType for v8::Value', C:\Users\runneradmin\.cargo\registry\src\github.com-1ecc6299db9ec823\serde_v8-0.9.0\src\payload.rs:32:5
```

I'm not experienced enough in the Deno internals or Rust to take this on myself, but I did some digging.
From that `bigint` panic, `payload.rs` in serde_v8 does not have an entry for BigInt. I found a related commit denoland/serde_v8@57ce430716d8abcd44a10166b3282bdb66703f78, but this related issue is still unresolved denoland/deno#6169.

It also feels like the panic error message in [extensions/ffi/lib.rs](https://github.com/denoland/deno/blob/v1.13.0/extensions/ffi/lib.rs) could be improved to show what type was attempted to be passed (if possible to help with issues like this), but I get this API is still unstable and very, very new.

### PR DESCRIPTION
**Pull request by eliassjogreen**
Fixes #11641 and implements #14521. This changes it so that all integers can also be passed as `number | bigint`, internally using the `U32x2` type for conversions. All native types over 32 bits (`u64`, `i64`, `usize`, `isize`) now result in a `bigint` being returned.

### CODE REVIEW

**Comment by aapoalas** [comment]
On a related note, why was the `U32x2` type needed again? Is there no way to handle the conversion cleanly at serde_v8 level from `v8::Local<v8::Value>` standing for a BigInt into `u64`, `i64` or `f64` directly?

**Comment by bartlomieju** [comment]
@littledivy please review

File: cli/dts/lib.deno.unstable.d.ts [hunk 1]
```diff
@@ -360,17 +360,25 @@ declare namespace Deno {
   }
 
   /** All possible number types interfacing with foreign functions */
-  type StaticNativeNumberType = Exclude<NativeType, "void" | "pointer">;
+  type StaticNativeNumberType = Exclude<
+    NativeType,
+    "void" | "u64" | "i64" | "usize" | "isize" | "pointer"
+  >;
```
Comment (littledivy) [inline cli/dts/lib.deno.unstable.d.ts:366]:
```suggestion
  type StaticNativeNumberType = Exclude<
    NativeType,
    "void" | "pointer" | StaticNativeBigIntType
  >;
```

**Review by littledivy — COMMENTED** [review]
lgtm, can you rebase? that should fix the rustc cache keys issue

**Pull request merged**